File: Doodream/소수 찾기.js
function solution(numbers) {
    numbers = numbers.split('');
    const isPrime = (n) => {
        if (n <= 1) return false;
        if (n === 2 || n === 3) return true;
        if (n % 2 == 0) return false;
        var divider = 3;
        while (n / 2 > divider) {
            if (n % divider === 0) return false;
            divider += 2;
        }
        return true;
    }
    var pickNumberArr = [];
    const permutation = (arr, n, k) => {
        if (n === arr.length - 1) {
            const str = arr.slice(0, k).join("");
            if (!pickNumberArr.includes(str)) pickNumberArr.push(str);
            return;
        } else {
            for (let i = 0; i < arr.length; i++) {
                var tmp = arr[n];
                arr[n] = arr[i];
                arr[i] = tmp;

                permutation(arr, n + 1, k);
                arr[i] = arr[n];
                arr[n] = tmp;
            }
            return pickNumberArr;
        }
    };
    for (let k = 0; k < numbers.length; k++) {
        permutation(numbers, 0, k + 1);
    }
    var tmp = [];
    for (let i = 0; i < pickNumberArr.length; i++) {
        tmp[i] = Number(pickNumberArr[i]);
    }
    var set = new Set(tmp);
    pickNumberArr = [...set];
    //pickNumberArr.filter(n => isPrime(n) ? true : false);
    var answer = 0;
    pickNumberArr.forEach(n => {
        isPrime(n) ? answer++ : answer;
    })

    return answer;
}
var numbers = "17";
console.log(solution(numbers));

File: yeonkyung/소수 찾기.py
import math
def isPrime(n):
    if n < 2:
        return False
    for i in range(2, int(math.sqrt(n+1)) + 1):
        if n % i == 0:
            return False
    return True

def solution(numbers):
    numbers = list(numbers)
    tmp = [-1] * len(numbers)
    ch = [0] * len(numbers)
    answer = []

    def dfs(idx):
        if idx == len(numbers):
            return answer

        for i in range(len(numbers)):
            if ch[i] == 0:
                ch[i] = 1
                tmp[idx] = int(numbers[i])
                answer.append(int(''.join([str(n) for n in tmp if n != -1])))
                dfs(idx + 1)
                ch[i] = 0
                tmp[idx] = -1

    dfs(0)
    count = 0
    for number in set(answer):
        if isPrime(number):
            count += 1
    return count

if __name__ == "__main__":
    numbers = "17"
    print(solution(numbers))